scene/resources/resource_format_text.cpp

```diff
@@ -907,6 +907,84 @@ Error ResourceLoaderText::rename_dependencies(Ref<FileAccess> p_f, const String
 	return OK;
 }
 
+// Fetches the script attached to a custom resource, reading as few lines as possible from the resource.
+// Please call ResourceLoaderText::open() before calling this.
+Error ResourceLoaderText::get_resource_script_path(String &r_script_path) {
+	if (res_type != "Resource") {
+		error_text = "Expected a user-defined resource file";
+		_printerr();
+		error = ERR_PARSE_ERROR;
+		return error;
+	}
```

Comment:
Is the direct comparison to "Resource" correct/intentional? Like, does it seem like the `ResourceLoaderText` only applies to resources that extend the base `Resource` type, or could there be C++ types deriving `Resource` that still save as generic resource file types? I'm not sure. Just want to double-check.

Reply:
Just did some testing and it looks like the res_type is stored as whatever base type your script extends from. I've only extended resources in my use cases, so this never occurred to me.